system/httplib/httplib.h

```diff
@@ -3620,6 +3624,7 @@ template <typename... Args>
 inline ssize_t Stream::write_format(const char *fmt, const Args &... args) {
   const auto bufsiz = 2048;
   std::array<char, bufsiz> buf;
+  buf[0]=0;
```

Comment:
question: what was the error on this one?

as far as I can see, it should not be necessary, the sprintf's will null terminate.

If it's a false positive and this is a workaround to keep the compiler happy only,
needs a comment to say what the issue is.

Reply:
@jakesmith In both cases the error was this. I will add comments

/mnt/e/HPCC/HPCC-Platform/system/httplib/httplib.h:3629:42: error: ‘buf’ may be used uninitialized [-Werror=maybe-uninitialized]
3629 |   auto sn = snprintf(buf.data(), buf.size() - 1, fmt, args...);

Reply:
@ghalliday @richardkchapman  - is this a case of the compiler being overzealous?

In cases like this, passing an uninitialized variable/array to a function, which is supposed to fill it, it may not be able to tell that sprintf is a function designed to fill it,
but to have explicit initialization ahead of a function call that initializes, seems ott.

Reply:
If we have to initialize, I think:
```
std::array<char, bufsiz> buf{""}
```

would be more elegant.